packages/pigeon/lib/swift_generator.dart

```diff
@@ -605,8 +605,7 @@ import FlutterMacOS
             'nullable enums require special code that this helper does not supply');
         return '${_swiftTypeForDartType(type)}(rawValue: $value as! Int)!';
       } else if (type.baseName == 'Object') {
-        // Special-cased to avoid warnings about using 'as' with Any.
-        return value;
+        return value + (type.isNullable ? '' : '!');
```

Comment:
This is to fix the bug that silences type mismatch. more details: https://github.com/flutter/packages/pull/3658#issuecomment-1526846509